unity-renderer/Assets/Scripts/MainScripts/DCL/AvatarSystem/AvatarWithHologram.cs

```diff
@@ -113,9 +111,20 @@ public async UniTask Load(List<string> wearablesIds, AvatarSettings settings, Ca
             }
         }
 
-        public void AddVisibilityConstrain(string key) { visibility.AddGlobalConstrain(key); }
+        public void AddVisibilityConstrain(string key)
+        {
+            visibility.AddGlobalConstrain(key);
+            baseAvatar.ToggleHideAvatarArea(key.Equals("IN_HIDE_AREA"));
+        }
 
-        public void RemoveVisibilityConstrain(string key) { visibility.RemoveGlobalConstrain(key); }
+        public void RemoveVisibilityConstrain(string key)
+        {
+            visibility.RemoveGlobalConstrain(key);
+            if (key.Equals("IN_HIDE_AREA"))
```

Comment:
Does it matter? Particles must be disabled regarding the source that hides the avatar.

I would disable the particles if the avatar is not visible (regarding the source). You can implemente a new set of methods to consult the status on visibility.
``` 
IVisibility.IsGloballyVisible()
IVisibility.IsMainRendererVisible()
IVisibility.AreFacialFeatureVisible()
```

And then I would just disable particles if `IsGloballyVisible == false` and enable them otherwise.